### DISCUSSION TITLE
Command documentation

### DISCUSSION THREAD

**Discussion by neur0manc** [NeoVintageous/NeoVintageous#987] [Q&A] [answered]
Is there comprehensive documentation about internal commands from NeoVintageous and Sublime Text? I have noticed a few examples like these:

```
nnoremap FileType go,html gd :LspSymbolDefinition<CR>
nnoremap ,f :ShowOverlay overlay=goto text=@<CR>
nnoremap <D-i> :GotoSymbolInProject<CR>
```

But I have not found any documentation for these examples, like
- What other overlays can I trigger with `:ShowOverlay`? `goto` and what else?
- What other commands are available, aside from `ShowOverlay` and `GotoSymbolInProject`?
- Can I generate a list of all commands that are available, including those from plugins (like `LspSymbolDefinition`)?

I'd be happy to help with the README once I have got a grasp on how this is structured, I'm sure it is useful information.

**Comment by gerardroche** [comment, marked as answer]
I updated the docs a little:


### Mapping Sublime Text Commands

Sublime Text commands are mappable. The command must be PascalCase and the parameters must be space separated.

Note: Sublime Text commands are PascalCased for the purpose of distinguishing them from Ex commands, which always start with a lowercase letter. This choice is due to an implementation detail.

**Example:** Map `,f` to the `show_overlay` command with arguments `{"overlay": "goto", "text": "@"}`

```vim
nnoremap ,f :ShowOverlay overlay=goto text=@<CR>
```

**Where do these commands come from?**

When you run a command in Sublime Text, you can inspect the console log to discover the command and arguments needed to map it.

1. Console Logging: View console logging by navigating to `Menu → View → Show Console`.

2. Command and Input Logging: Enable command and input logging by running the following commands in the console:

   ```
   sublime.log_commands(True)
   sublime.log_input(True)
   ```

For example, let's say you want to map "Goto Symbol" command above. Run the command, e.g., "Menu → Goto → Goto Symbol" command and Sublime Text will print the following in the console:

```
command: show_overlay {"overlay": "goto", "text": "@"}
```


## Blog post

- [Mapping Sublime Text commands](https://blog.gerardroche.com/2023/05/16/neovintageous-mapping-sublime-text-commands/)

Basically you can inspect the details of commands that run via console logging. Once you have that information you can map it. This includes third party plugins. For example, I provide examples in some of my plugins, but even if a plugin author doesn't provide examples mappings, you can still run the command and inspect it via the console.

**Examples**:

- PHPUnitKit - https://github.com/gerardroche/sublime-phpunit
- GitOpen - https://github.com/gerardroche/sublime-git-open
- Gitk - https://github.com/gerardroche/sublime-gitk

**Reply by gerardroche** [reply]
For a full list of supported ex commands, please refer to our [roadmap](https://github.com/NeoVintageous/NeoVintageous/blob/master/ROADMAP.md).

**Comment by neur0manc** [comment]
That is absolutely excellent @gerardroche, exactly what I was looking for. The console logging tip is gold. Thank you very much!
